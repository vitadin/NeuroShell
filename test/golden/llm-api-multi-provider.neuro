%% Test LLM API commands with multiple provider scenarios using \set-env
%% This test demonstrates the complete workflow with test API keys

%% Step 1: Set up test API keys using the new \set-env command
\echo ═══ Setting Up Test API Keys ═══
\set-env[OPENAI_API_KEY=sk-test1234567890abcdef1234567890abcdef]
\set-env[ANTHROPIC_API_KEY=ant-test1234567890abcdef1234567890abcdef]
\set-env[MOONSHOT_API_KEY=moon-test1234567890abcdef1234567890abcdef]

%% Step 2: Load all API keys (should now find the test keys)
\echo ═══ Loading All API Keys ═══
\llm-api-load

%% Step 3: Test provider-specific filtering
\echo ═══ Testing Provider-Specific Filtering ═══
\llm-api-load[provider=openai]
\llm-api-load[provider=anthropic]
\llm-api-load[provider=moonshot]

%% Step 4: Activate keys for different providers
\echo ═══ Activating OpenAI Key ═══
\llm-api-activate[provider=openai, key=os.OPENAI_API_KEY]

\echo ═══ Activating Anthropic Key ═══  
\llm-api-activate[provider=anthropic, key=os.ANTHROPIC_API_KEY]

%% Step 5: Load final state with active keys
\echo ═══ Final State with Active Keys ═══
\llm-api-load

%% Step 6: Test activation with moonshot
\echo ═══ Activating Moonshot Key ═══
\llm-api-activate[provider=moonshot, key=os.MOONSHOT_API_KEY]

%% Step 7: Verify active key variables are set correctly
\echo ═══ Active Key Variables ═══
\get[#active_openai_key]
\get[#active_anthropic_key]
\get[#active_moonshot_key]