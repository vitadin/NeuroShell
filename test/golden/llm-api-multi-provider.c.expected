‚ïê‚ïê‚ïê Setting Up Test API Keys ‚ïê‚ïê‚ïê
Setting OPENAI_API_KEY = sk-test1234567890abcdef1234567890abcdef
Setting ANTHROPIC_API_KEY = ant-test1234567890abcdef1234567890abcdef
Setting MOONSHOT_API_KEY = moon-test1234567890abcdef1234567890abcdef
‚ïê‚ïê‚ïê Loading All API Keys ‚ïê‚ïê‚ïê

  # LLM API Keys Found                                                        
                                                                              
   Variable Name          | API Key                | Status                   
  ------------------------|------------------------|------------------------  
   os.ANTHROPIC_API_KEY   | ant...def              | INACTIVE                 
   os.MOONSHOT_API_KEY    | moo...def              | INACTIVE                 
   os.OPENAI_API_KEY      | sk-...def              | INACTIVE                 

‚ïê‚ïê‚ïê Testing Provider-Specific Filtering ‚ïê‚ïê‚ïê

  # LLM API Keys Found (Openai only)                                          
                                                                              
   Variable Name          | API Key                | Status                   
  ------------------------|------------------------|------------------------  
   os.OPENAI_API_KEY      | sk-...def              | INACTIVE                 


  # LLM API Keys Found (Anthropic only)                                       
                                                                              
   Variable Name          | API Key                | Status                   
  ------------------------|------------------------|------------------------  
   os.ANTHROPIC_API_KEY   | ant...def              | INACTIVE                 


  # LLM API Keys Found (Moonshot only)                                        
                                                                              
   Variable Name          | API Key                | Status                   
  ------------------------|------------------------|------------------------  
   os.MOONSHOT_API_KEY    | moo...def              | INACTIVE                 

‚ïê‚ïê‚ïê Activating OpenAI Key ‚ïê‚ïê‚ïê
‚úì Activated os.OPENAI_API_KEY API key for openai provider
  ${#active_openai_key} now contains the API key value
‚ïê‚ïê‚ïê Activating Anthropic Key ‚ïê‚ïê‚ïê
‚úì Activated os.ANTHROPIC_API_KEY API key for anthropic provider
  ${#active_anthropic_key} now contains the API key value
‚ïê‚ïê‚ïê Final State with Active Keys ‚ïê‚ïê‚ïê

  # LLM API Keys Found                                                        
                                                                              
   Variable Name          | API Key                | Status                   
  ------------------------|------------------------|------------------------  
   os.ANTHROPIC_API_KEY   | ant...def              | **üü¢ ACTIVE**            
   os.MOONSHOT_API_KEY    | moo...def              | INACTIVE                 
   os.OPENAI_API_KEY      | sk-...def              | **üü¢ ACTIVE**            
                                                                              
  **Active keys:** ${#active_openai_key}, ${#active_anthropic_key}            

‚ïê‚ïê‚ïê Activating Moonshot Key ‚ïê‚ïê‚ïê
‚úì Activated os.MOONSHOT_API_KEY API key for moonshot provider
  ${#active_moonshot_key} now contains the API key value
‚ïê‚ïê‚ïê Active Key Variables ‚ïê‚ïê‚ïê
#active_openai_key = sk-test1234567890abcdef1234567890abcdef
#active_anthropic_key = ant-test1234567890abcdef1234567890abcdef
#active_moonshot_key = moon-test1234567890abcdef1234567890abcdef